content/v2.1/helpers/overview.md

```diff
@@ -3,8 +3,231 @@ title: Overview
+
+Given you have a view exposure defined:
+
+```ruby
+# app/views/books/show.rb
+
+module Bookshelf
+  module Views
+    module Books
+      class Show < Bookshelf::View
+        expose :fancy_header do
+          "Fancy header"
+        end
+      end
+    end
+  end
+end
+```
+
+Then you can have the part with the helper used.
+
+```ruby
+# app/views/parts/fancy_header.rb
+
+module Bookshelf
+  module Views
+    module Parts
+      module FancyHeader
+        def content
+          helpers.tag.h1(value)
+        end
+      end
+    end
+  end
+end
+```
+
+Then in the template you can just access the part:
+
+```ruby
+# app/templates/books/show.html.erb
+
+<%= fancy_header %>
+```
+
+### Features
+
+Here are lists of features with the corresponding examples:
+
+**Auto-closing tags according to HTML5 spec**
+
+```ruby
+tag.div # => <div></div>
+```
+
+**Content string as first argument**
+
+```ruby
+tag.img # => <img>
+tag.div("First argument is content")        # => <div>First argument is content</div>
+```
+
+**Accepts content as block returning a string**
+
+```ruby
+tag.div { "Content in the block" }     # => <div>Content in the block</div>
+```
+
+**Allows for nesting tags via argument or block**
+
+```ruby
+tag.div(tag.p("Nested tag builder")) # => <div><p>Nested tag builder</p></div>
+
+tag.div do
+  tag.p("Also nested tag builders.")
+end
+```
+
+**It builds attributes from given hash**
+
+```ruby
+tag.div("With Additional attributes", class: ["a", "b"]) # => <div class="a b">With additional attributes</div>
+```
+
+**Builds nested HTML attributes attributes from given hash**
+
+```ruby
+tag.div(id: "el", data: {x: "y"}) # => <div id="el" data-x="y"></div>
+tag.div(id: "el", aria: {x: "y"}) # => <div id="el" aria-x="y"></div>
+```
+
+**Allows for dynamic attributes visibility control**
+
+```ruby
+tag.div("Dynamic attributes", class: {"a": true, "b": false}) # => <div class="a">Dynamic Attributes</div>
+```
+
+**Supports custom tags**
+
+```ruby
+tag.custom_tag("hello") # => <custom-tag>hello</custom-tag>
+```
+
+
+### Escaping HTML input
+
+The tag contents are automatically escaped for security reasons:
+
+```ruby
+tag.p("<script>alert()</script>")         # => <p>&lt;script&gt;alert()&lt;/script&gt;</p>
+tag.p(class: "<script>alert()</script>")  # => <p class="&lt;script&gt;alert()&lt;/script&gt;"></p>
+```
+
+To bypass it use `html_safe` on the given content
+
+```ruby
+tag.p("<em>safe content</em>".html_safe)  # => <p><em>safe content</em></p>
+```
+
+### Standalone use hanami-view
+
+When using hanami-view standalone, include this module directly in your base part and scope
+
+```ruby
+class BasePart < Hanami::View::Part
+  include Hanami::View::Helpers::TagHelper
+end
+
+class BaseScope < Hanami::View::Scope
+  include Hanami::View::Helpers::TagHelper
+end
+
+class BaseView < Hanami::View
+  config.part_class = BasePart
+  config.scope_class = BaseScope
+end
+```
+
+## link_to helper
```

Comment:
@swilgosz What's your idea for organizing all the helpers' documentation? All in here or separated docs?

Reply:
At first I thought they'll fit in in a single document, but after working on form_for, that's not gonna happen for sure. I can extract to a separate doc, same as for others.